Implement SameSite Support (Fixes #1)

SlvrEagle23 opened the pull request:
A number of browsers are starting to warn that setting a cookie with a SameSite value of "None" without "Secure" won't be supported in the future, but since this is the default if you use the current default parameters with this library, I found it imperative to implement support for setting the same-site policy manually, while also having it default to the newer more broadly appropriate "Lax" setting.

Feel free to modify as needed. One important note is that this requires version 2.0.0 or newer of the SetCookie library, as previous versions don't support setting the SameSite value at all. The dependencies in composer.json have been updated accordingly to reflect this.

SlvrEagle23 commented:
Another note: without supporting the 1.0.x branch of SetCookie, one also loses support for PHP 7.1, meaning only 7.2 or newer is supported by the whole package by changing that requirement.

One could also take the alternate approach of simply _not setting_ the "SameSite" value (or appending it by hand onto the end of the cookie string), but both of those approaches undermine the utility of the SetCookie library itself and/or yield inconsistent results.

Personally, I'm not too bothered by the notion of removing PHP 7.1 compatibility (what with it being EOL and what not), though I understand this likely means a new semver major release.

michalbundyra commented on composer.json:
```diff
@@ -31,8 +31,8 @@
-        "php": "^7.1",
-        "dflydev/fig-cookies": "^1.0.2 || ^2.0",
+        "php": "^7.2",
+        "dflydev/fig-cookies": "^2.0",
```
We must use 2.0.1 version, as it contains the key fix: see https://github.com/dflydev/dflydev-fig-cookies/compare/v2.0.0...v2.0.1

michalbundyra commented on composer.json:
```diff
@@ -31,8 +31,8 @@
     "require": {
-        "php": "^7.1",
-        "dflydev/fig-cookies": "^1.0.2 || ^2.0",
+        "php": "^7.2",
```
We are fine with bumping the PHP required version. PHP 7.2 is in security only and in all other libraries where we provided SameSite cookie directive support we have bumped PHP to 7.3 - see mezzio-swoole or mezzio-session-ext (https://github.com/mezzio/mezzio-session-ext/pull/9/files).

So I am suggesting to use PHP 7.3 here as well :)

SlvrEagle23 commented:
@michalbundyra I've rebased against develop and updated the PR to be against develop, bumped the version dependency to 7.3 and added support for the persistencefactory to also use same-site config, and updated several of the unit tests to check this value as well.

I'm not completely comfortable enough with the Laminas way of writing unit tests to actually create brand new ones, but if anyone is and would be willing to fill that in, I think that would cover everything.

weierophinney commented:
Thanks, @SlvrEagle23!
